graphqa/predict.py

```diff
@@ -92,13 +93,46 @@ def download_model(args):
         with zipfile.ZipFile(zip_path,"r") as zip_ref:
             zip_ref.extractall(args["model_dir"])
 
+
+def load_questions(args):
+    """ Read in the predefined questions. """
+    with open(args["questions_path"]) as fp:
+        questions = fp.read().splitlines()
+        print('%d questions loaded, avg. len of %d' % (len(questions), np.mean([len(d.split()) for d in questions])))
+        return questions
+
+def cos_sim(a, b):
+    """
+    Takes 2 vectors a, b and returns the cosine similarity according to the
+    definition of the dot product.
+    """
+    dot_product = np.dot(a, b)
+    norm_a = np.linalg.norm(a)
+    norm_b = np.linalg.norm(b)
+    return dot_product / (norm_a * norm_b)
+
+
+def extract_one_bert(query, questions):
+    """
+    User BERT to find the most similar question from the questions. This
+    can also be realized by fuzzywuzzy.
+    BERT Server needs to be started first. See start-bert-server.sh.
+    """
+    bc = BertClient(port=5555, port_out=5556)
```

Comment:
Have you imported `BertClient`?

Reply:
Perhaps, should initialize `BertClient` as global, then call it when needed.

Reply:
The BertClient import is now fixed. Indeed, the BertClient call does not belong here. This is a simple dirty way of testing BERT encoder. If one wants to use BERT for encoding the question and use BertServer to serve the model, one may also need to specify the ip addres of the BertServer and tweak some other configurations. I will leave these for later. Also, the function "extract_one_bert" is commented out and not being called in the current code.